lib/tableau_api/connection.rb

```diff
@@ -3,28 +3,28 @@ class Connection
     API_VERSION = '3.1'.freeze
 
     include HTTParty
+    headers 'User-Agent' => "tableau_api/#{::TableauApi::VERSION} Ruby/#{RUBY_VERSION}"
 
     def initialize(client)
       @client = client
     end
 
-    def post(path, *args)
-      self.class.post("#{@client.host}/#{path}", *args)
+    def post(path, **kwargs)
+      self.class.post("#{@client.host}/#{path}", kwargs)
     end
 
     # if the result is paginated, it will fetch subsequent pages
     # collection can be delimited with a period to do nested hash lookups
     # e.g. objects.object
-    def api_get_collection(path, collection, *args)
+    def api_get_collection(path, collection, page_number: 1, page_size: 100, **kwargs)
       Enumerator.new do |enum|
-        args[0] = {} unless args[0]
-        page_size = (args[0].delete(:page_size) { 100 }).to_i
-        page_number = (args[0].delete(:page_number) { 1 }).to_i
-
         loop do
-          uri = URI::HTTP.build(path: "/#{path}", query: URI.encode_www_form(**args[0], pageSize: page_size, pageNumber: page_number)).request_uri
+          query = kwargs.fetch(:query, '')
```

Comment:
So does this only support passing query as a string? Does httparty generally support a hash? But then it does tableau-incompatible URL encoding?

Reply:
That's correct on all counts.